uint16 query_id
uint8 length
NeighbourNode[] vec